reference/configuration/framework.rst

```diff
@@ -169,39 +293,431 @@ see :doc:`/cookbook/request/load_balancer_reverse_proxy`.
+
+You can also set ``esi`` to ``true`` to enable it:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # app/config/config.yml
+        framework:
+            esi: true
+
+    .. code-block:: xml
+
+        <!-- app/config/config.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:framework="http://symfony.com/schema/dic/symfony"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/symfony http://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
+
+            <framework:config>
+                <esi />
+            </framework:config>
+        </container>
+
+    .. code-block:: php
+
+        // app/config/config.php
+        $container->loadFromExtension('framework', array(
+            'esi' => true,
+        ));
+
+fragments
+~~~~~~~~~
+
+.. seealso::
+
+    Learn more about fragments in the
+    :ref:`HTTP Cache article <book-http_cache-fragments>`.
+
+.. _reference-fragments-enabled:
+
+enabled
+.......
+
+**type**: ``boolean`` **default**: ``false``
+
+Whether to enable the fragment listener or not. The fragment listener is used
+to render ESI fragments independently of the rest of the page.
+
+This setting is automatically set to ``true`` when one of the child settings is
+configured.
+
+.. _reference-fragments-path:
+
+path
+....
+
+**type**: ``string`` **default**: ``'/_fragment'``
+
+The path prefix for fragments. The fragment listener will only be executed
+when the request starts with this path. It becomes the service container
+parameter called ``fragment.path``.
+
+profiler
+~~~~~~~~
+
+.. _reference-profiler-enabled:
+
+enabled
+.......
+
+.. versionadded:: 2.2
+    The ``enabled`` option was introduced in Symfony 2.2. Previously, the profiler
+    could only be disabled by omitting the ``framework.profiler`` configuration
+    entirely.
+
+**type**: ``boolean`` **default**: ``true`` in the ``dev`` and ``test`` environments
+
+The profiler can be disabled by setting this option to ``false``.
+
+.. note::
+
+    The profiler works independently from the Web Developer Toolbar, see the
+    :doc:`WebProfilerBundle configuration
+    </reference/configuration/web_profiler>` on how to disable/enable it.
+
+collect
+.......
+
+.. versionadded:: 2.3
+    The ``collect`` option was introduced in Symfony 2.3. Previously, when
+    ``profiler.enabled`` was ``false``, the profiler *was* actually enabled,
+    but the collectors were disabled. Now, the profiler and the collectors
+    can be controlled independently.
+
+**type**: ``boolean`` **default**: ``true``
+
+This option configures the way the profiler behaves when it is enabled. If set
+to ``true``, the profiler collects data for all requests. If you want to only
```

Comment:
... collects data for all requests ...

This is true, unless you change one of the options below - most notably `matcher`. In that case, it will only collect when the `matcher` says to collect. If you set this to false, it *is* true that the collector is totally disabled though.